internal/pkg/service/buffer/watcher/apinode/state.go

```diff
@@ -101,23 +101,26 @@ func (s *state) onError(err error) {
 func watch[T fmt.Stringer](ctx context.Context, wg *sync.WaitGroup, s *state, prefix etcdop.PrefixT[T], rev *revision.Syncer) *stateOf[T] {
 	tree := prefixtree.New[T]()
 
-	ch, _initDone := prefix.GetAllAndWatch(ctx, s.client, s.onError, etcd.WithCreatedNotify(), etcd.WithPrevKV())
+	initDone := make(chan error)
+	ch := prefix.GetAllAndWatch(ctx, s.client, s.onError, etcd.WithCreatedNotify(), etcd.WithPrevKV())
 
 	// Log only changes, not initial load
 	logsEnabled := atomic.NewBool(false)
-	initDone := make(chan error)
-	go func() {
-		err := <-_initDone
-		logsEnabled.Store(true)
-		initDone <- err
-	}()
```

Comment:
A tu su zmeny v kode, ... ako mozes vidiet, tak tym, ze je to spravne, je to aj kratsie.

Reply:
`logsEnabled.Store` vieme obsluzit uz vo `for`.